WcaOnRails/app/controllers/admin_controller.rb

```diff
@@ -291,43 +291,54 @@ def edit_person
   end
 
   def update_person
-    @person = Person.current.find_by(wca_id: params[:person][:wca_id])
-    if @person
-      person_params = params.require(:person).permit(:name, :countryId, :gender, :dob, :incorrect_wca_id_claim_count)
+    wca_id = params.require(:person).permit(:wcaId)[:wcaId]
+    person = Person.current.find_by(wca_id: wca_id)
+    success_message = nil
+    warning_message = nil
+    error_message = nil
+    if person
+      person_params = params.require(:person).permit(:name, :gender, :dob)
+      person_params[:countryId] = Country.find_by_iso2(params[:person][:representing]).id
```

Comment:
You are very inconsistent about how you're reading params.
Especially handling it like `params.require(:person).permit(:wcaId)[:wcaId]` is horrible. Please find a consistent way to `params.require(:person)` **once**, and then read from that `require`d hash

Reply:
Okay, I cleaned it a bit. Hope it's fine now.